src/Eloquent/HybridRelations.php

```diff
@@ -221,7 +221,7 @@ public function morphTo($name = null, $type = null, $id = null, $ownerKey = null
                 $this->newQuery(),
                 $this,
                 $id,
-                $ownerKey,
+                $ownerKey ?: $this->getKeyName(),
```

Comment:
https://github.com/mongodb/laravel-mongodb/issues/2783 suggests that this broke MorphTo relationships when a model uses a custom primary key.

https://github.com/mongodb/laravel-mongodb/pull/3011 includes a proposed fix.